### PR TITLE
Refactor: Created new Use Case: Add Location

### PR DESCRIPTION
**Pull request by chickenwaddle77**
Refactored old code from 'Location' Class to separate Clean Architecture components. Separated responsibilities for methods into different classes. Added abstract layer (interfaces) to make the code more modular and independent. Separated previous 'Location' Class to the Use Case: Add Location and Data Access: FourSquareDataAccess. Added ConfigLoader as apart of Data Access. Yet to connect Controller to this Use Case.

### CODE REVIEW

File: src/main/java/data_access/ConfigLoaderDataAccess.java
Comment (a1-su) [inline src/main/java/data_access/ConfigLoaderDataAccess.java:1]: ConfigLoaderDataAccess has already been implemented as ConfigLoader in data_access. Could you replace instances where you call ConfigLoaderDataAccess with the current ConfigLoader class we're using?

File: src/main/java/data_access/FoursquareDataAccess.java
Comment (a1-su) [inline src/main/java/data_access/FoursquareDataAccess.java:1]: Nit: Rename to FoursquareLocationProvider

File: src/main/java/data_access/GoogleMapsLocationProvider.java [hunk 1]
```diff
@@ -17,7 +17,7 @@
 public class GoogleMapsLocationProvider {
 
     private final OkHttpClient client = new OkHttpClient();
-    private final String apiKey = ConfigLoader.getKey("google.api.key");
+    private final String apiKey = ConfigLoaderDataAccess.getKey("google.api.key");
```
Comment (a1-su) [inline src/main/java/data_access/GoogleMapsLocationProvider.java]: Nit: Use the old ConfigLoader, not ConfigLoaderDataAccess

**Review by a1-su — CHANGES_REQUESTED** [review]

**Review by a1-su — APPROVED** [review]
LGTM!

**Comment by chickenwaddle77** [comment]
Thank you

**Pull request merged**